boards/shared/Src/Io/Io_SharedSpi.c

```diff
@@ -0,0 +1,114 @@
+#include <assert.h>
+#include <stdlib.h>
+#include "Io_SharedSpi.h"
+
+#define RETURN_IF_STATUS_NOT_OK(status, nss_port, nss_pin)        \
+    if ((status) != HAL_OK)                                       \
+    {                                                             \
+        HAL_GPIO_WritePin((nss_port), (nss_pin), GPIO_PIN_RESET); \
+        return (status);                                          \
+    }
+
+struct SharedSpi
+{
+    SPI_HandleTypeDef *hspi;
+    GPIO_TypeDef *     nss_port;
+    uint16_t           nss_pin;
+};
+
+struct SharedSpi *Io_SharedSpi_Create(
+    SPI_HandleTypeDef *hspi,
+    GPIO_TypeDef *     nss_port,
+    uint16_t           nss_pin)
+{
+    assert(hspi != NULL);
+
+    struct SharedSpi *shared_spi = malloc(sizeof(struct SharedSpi));
+    assert(shared_spi != NULL);
+
+    shared_spi->hspi     = hspi;
+    shared_spi->nss_pin  = nss_pin;
+    shared_spi->nss_port = nss_port;
+
+    return shared_spi;
+}
+
+void Io_SharedSpi_SetNssLow(const struct SharedSpi *const spi)
+{
+    HAL_GPIO_WritePin(spi->nss_port, spi->nss_pin, GPIO_PIN_RESET);
+}
+
+void Io_SharedSpi_SetNssHigh(const struct SharedSpi *const spi)
+{
+    HAL_GPIO_WritePin(spi->nss_port, spi->nss_pin, GPIO_PIN_SET);
+}
+
+HAL_StatusTypeDef Io_SharedSpi_TransmitAndReceive(
+    const struct SharedSpi *const spi,
+    uint8_t *                     tx_data,
+    uint16_t                      tx_size,
+    uint8_t *                     rx_data,
+    uint16_t                      rx_size)
+
+{
+    Io_SharedSpi_SetNssLow(spi);
+    RETURN_IF_STATUS_NOT_OK(
```

Comment:
Felt I was repeating code here when I'm checking for the HAL status returned after each SPI transmission/reception so I decided to use a macro here